File: DesignPatternsExercises/java/SingletonDesignPattern/Exercise/SingletonDesignPattern.java
public class SingletonDesignPattern {

	public static void main(String[] args) {
		FamilyMember father = new FamilyMember("Father");
		System.out.println("Current family balance: $" + father.getBalance());
		System.out.println("Father deposits $1000");
		father.deposit(1000);
		System.out.println("Balance after deposit: $" + father.getBalance());

		FamilyMember son = new FamilyMember("Son");
		System.out.println("Son withdraws $200");
		son.withdraw(200);
		System.out.println("Balance after withdrawal: $" + son.getBalance());

		System.out.println("Son withdraws $1000");
		son.withdraw(1000);
		System.out.println("Balance after withdrawal: $" + son.getBalance());
	}

}

File: DesignPatternsExercises/ruby/singleton_design_pattern/solution/singleton_design_pattern.rb
require_relative 'family_member'

father = FamilyMember.new(:Father)
puts "Current family balance: $#{father.balance}"
puts "\nFather deposits $1000"
father.deposit(1000)
puts "Balance after deposit: $#{father.balance}"

son = FamilyMember.new(:Son)
puts "\nSon withdraws $200"
son.withdraw(200)
puts "Balance after withdrawal: $#{son.balance}"

puts "\nSon withdraws $1000"
son.withdraw(1000)
puts "Balance after withdrawal: $#{son.balance}"
